### DISCUSSION TITLE
Data Table focus state

### DISCUSSION THREAD

**Discussion by half5cat** [carbon-design-system/carbon#9799] [Help] [answered]
Is this intentional? If so, please elaborate on this.

Focus on Expandable Icon:
![image](https://user-images.githubusercontent.com/1887866/135988683-8aea2b17-1937-480e-b830-0226d3f2e3d7.png)

Focus on Checkbox:
![image](https://user-images.githubusercontent.com/1887866/135988910-031a98ac-3e87-45d9-8ab0-23484f4d842e.png)

Source for the above: https://react.carbondesignsystem.com/?path=/story/components-datatable-development--example 

Focus on Checkmark Column:
![image](https://user-images.githubusercontent.com/1887866/135989069-4c112814-3559-41cc-8668-58b2dad7a1fb.png)

Source: https://react.carbondesignsystem.com/?path=/story/components-datatable--with-checkmark-columns

**Comment by jnm2377** [comment, marked as answer]
Hi @half5cat great question! The first 2 images you shared are intentional. Those changes were made here: https://github.com/carbon-design-system/carbon/pull/8656#pullrequestreview-657196560 to have consistent focus states within data table.

<img width="252" alt="Screen Shot 2021-10-05 at 5 32 27 PM" src="https://user-images.githubusercontent.com/32556167/136112161-54ad209f-2d76-42c1-9dec-8f3126457d2d.png">
<img width="235" alt="Screen Shot 2021-10-05 at 5 32 21 PM" src="https://user-images.githubusercontent.com/32556167/136112163-5993e867-af05-4ebd-b925-bbac0003b16f.png">

The reason you're seeing the focus outline inconsistency, however, is because the select column checkbox is using data table specific selectors for those styles, while the second one is not.

#### data table select column (specific data table styles for that checkmark)
<img width="340" alt="Screen Shot 2021-10-05 at 5 33 06 PM" src="https://user-images.githubusercontent.com/32556167/136112145-27e764d2-ddf6-4999-804d-6a19be619e74.png">

#### just a checkmark inside data table body, hence not using specific data table styles
<img width="330" alt="Screen Shot 2021-10-05 at 5 31 09 PM" src="https://user-images.githubusercontent.com/32556167/136112130-15323531-a617-479d-8c1b-9934a51553f9.png">
 
You are right that these should be consistent. Looks like we should add a selector that works for any checkbox within data table, not just the select column.

**Reply by jnm2377** [reply]
I opened an issue for it: https://github.com/carbon-design-system/carbon/issues/9808

**Reply by half5cat** [reply]
Hey @jnm2377, thanks a lot for clarifying. However, I am also concerned about the difference in border radius.

**Reply by jnm2377** [reply]
Good point, I will add a comment on the issue to check with design on the border radius. 👍🏽
